### PR TITLE
Add a bunch of unit tests for trainer.py

### PR DESCRIPTION
**Pull request by bhancock8**
Primarily adds unit tests for trainer.py. Also:
- Drop unsupported optimizer/scheduler options (ones that were in the code that was ported over, but didn't have tests to show that they wouldn't actually be usable in their current state)
- Update "batch" -> "batches" and "epoch" -> "epochs" for consistent usage between lr_scheduler and LogManager.

**Test plan**:
New unit tests! These focus primarily on whether or not the appropriate optimizer/scheduler/checkpointer/etc. is found and initialized appropriately, not whether it functions as expected (which would be tested in the unit tests for the individual classes).

### CODE REVIEW

File: test/classification/training/test_trainer.py [hunk 1]
```diff
@@ -90,6 +91,94 @@ def test_trainer_errors(self):
...
+
+        trainer = Trainer(**trainer_config, optimizer="adam")
+        trainer.train_model(model, [dataloaders[0]])
+        self.assertIsInstance(trainer.optimizer, optim.Adam)
+
+        trainer = Trainer(**trainer_config, optimizer="adamax")
+        trainer.train_model(model, [dataloaders[0]])
+        self.assertIsInstance(trainer.optimizer, optim.Adamax)
+
+        with self.assertRaises(ValueError):
```
Comment (paroma) [inline test/classification/training/test_trainer.py]: For the `ValueErrors`, can we include checking the error message as a check for consistency
```python
with self.assertRaisesRegex(ValueError, "not currently implemented"):
```
